resources/views/kiosk/rfid.blade.php

```diff
@@ -32,10 +38,58 @@
 
 @section('customjs')
 <script>
-  $(document).ready(function(){
-      $(document).click(function() { $("#rfid").focus() });
-      $(document).mousedown(function() { $("#rfid").focus() });
-      $(document).mouseup(function() { $("#rfid").focus() });
-  });
+$(document).ready(function(){
+    $(document)
+        .click(function() { $("#rfid").focus() })
+        .mousedown(function() { $("#rfid").focus() })
+        .mouseup(function() { $("#rfid").focus() })
+        ;
+
+    // For developer reference, the following conditions must be met for the NDEFReader API to function:
+    //    - The user must be accessing the API through a supported browser. This generally only includes Android browsers like Chrome and WebView.
+    //    - The page must be served over HTTPS.
+    //    - The reader function must be initated by a user action -- usually a button press.
+    //    - Once the reader is initated, a modal asking for permission to access to NFC appears. This permission must be allowed.
+    // If any of these conditions are not met, the API will transparently fail and the functionality will be unavailable.
+
+    if ('NDEFReader' in window) { /* Scan and write NFC tags */
+        $('#scanButtonRow').show(); // Only show a scan button where the API is supported.
+
+        // For the Web NFC API to work, the reader must be triggered from a user action like this button.
+        // It does not work if you initiate the reader through a programatic function. :/
+        $('#scanButton').click(async () => {
+            $('#scanButton')
+              .removeClass('btn-success')
+              .addClass('btn-warning')
+              .text('Scanning...')
+              ;
+
+            console.log("User clicked scan button");
+
+            try {
+                const ndef = new NDEFReader();
+                await ndef.scan();
+                console.log("> Scan started");
+
+                ndef.addEventListener("readingerror", () => {
+                    console.log("Argh! Cannot read data from the NFC tag. Try another one?");
+                });
```

Comment:
It's interesting to me that the event listeners are added after the `await` rather than before